code/parachain/frame/staking-rewards/src/lib.rs

```diff
@@ -904,78 +914,135 @@ pub mod pallet {
 			who: &Self::AccountId,
 			(fnft_collection_id, fnft_instance_id): Self::PositionId,
 			amount: Self::Balance,
+			duration: DurationSeconds,
 			keep_alive: bool,
-		) -> Result<Self::PositionId, DispatchError> {
-			let mut stake = Stakes::<T>::get(fnft_collection_id, fnft_instance_id)
-				.ok_or(Error::<T>::StakeNotFound)?;
-			let mut rewards_pool = RewardPools::<T>::try_get(stake.reward_pool_id)
-				.map_err(|_| Error::<T>::RewardsPoolNotFound)?;
+		) -> DispatchResult {
+			Stakes::<T>::try_mutate(fnft_collection_id, fnft_instance_id, |maybe_stake| {
+				let stake = maybe_stake.as_mut().ok_or(Error::<T>::StakeNotFound)?;
 
-			let reward_multiplier = FixedU64::one().try_into_validated().expect(">= 1");
+				ensure!(duration >= stake.lock.duration, Error::<T>::CannotExtendToShorterDuration);
 
-			ensure!(
-				matches!(
-					T::Assets::can_withdraw(rewards_pool.asset_id, who, amount),
-					WithdrawConsequence::Success
-				),
-				Error::<T>::NotEnoughAssets
-			);
+				RewardPools::<T>::try_mutate(stake.reward_pool_id, |maybe_rewards_pool| {
```

Comment:
Indentation looks off

Reply:
looks fine to me, I know github has issues rendering indents sometimes though